ocaml/xapi/context.ml

```diff
@@ -503,12 +503,23 @@ let get_user_agent context =
 let with_tracing ?originator ~__context name f =
   let open Tracing in
   let parent = __context.tracing in
-  let span_attributes = Attributes.attr_of_originator originator in
+  let span_attributes =
+    Attributes.attr_of_originator originator
+    @ make_attributes ~task_id:__context.task_id
+        ?session_id:__context.session_id ()
+  in
   match start_tracing_helper ~span_attributes (fun _ -> parent) name with
-  | Some _ as span ->
+  | Some _ as span -> (
+    try
       let new_context = {__context with tracing= span} in
       let result = f new_context in
       let _ = Tracer.finish span in
       result
+    with exn ->
+      let backtrace = Printexc.get_backtrace () in
+      let error = (exn, backtrace) in
+      ignore @@ Tracer.finish span ~error ;
+      raise exn
```

Comment:
We could use `Printexc.raise_with_backtrace` here if we captured the raw backtrace instead at `let backtrace=` (it can be converted to string with another function from Printexc).
This will ensure that we also log the proper stacktrace in xensource.log.

Reply:
Done https://github.com/xapi-project/xen-api/pull/5735/commits/e89fd5ccb8bf6afa93b0ac29de6a03932e7bd60e